src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -0,0 +1,1977 @@
+            )
+        self._evaluate_node(
+            self.children[0].children[0],
+            private_list,
+            firstprivate_list,
+            shared_list,
+            in_list,
+            out_list,
+        )
+
+        # Make the clauses to return.
+        # We skip references to constants as we don't need them.
+        # Constants will never be private.
+        private_clause = OMPPrivateClause()
+        firstprivate_clause = OMPFirstprivateClause()
+        for ref in private_list:
+            # If the symbol is in the parallel_firstprivate set then
+            # we trust the parent parallel region and make it firstprivate
+            if ref.symbol in self._parallel_firstprivate:
+                firstprivate_list.append(ref)
+            else:
+                private_clause.addchild(ref)
+        for ref in firstprivate_list:
+            firstprivate_clause.addchild(ref)
+        shared_clause = OMPSharedClause()
+        for ref in shared_list:
+            shared_clause.addchild(ref)
+
+        in_clause = OMPDependClause(
+            depend_type=OMPDependClause.DependClauseTypes.IN
+        )
+        # For input references we need to ignore references to constant
+        # symbols. This means we need to try to get external symbols as well
+        for ref in in_list:
+            if isinstance(ref.symbol, DataSymbol) and ref.symbol.is_constant:
+                continue
+            if (ref.symbol.is_import and
+                    ref.symbol.get_external_symbol().is_constant):
+                continue
+            in_clause.addchild(ref)
+        out_clause = OMPDependClause(
+            depend_type=OMPDependClause.DependClauseTypes.OUT
+        )
+        for ref in out_list:
+            out_clause.addchild(ref)
+
+        return (
+            private_clause,
+            firstprivate_clause,
+            shared_clause,
+            in_clause,
+            out_clause,
+        )
+
+    def lower_to_language_level(self):
+        """
+        Lowers the structure of the PSyIR tree inside the Directive
+        to generate the Clauses that are required for this Directive.
+        """
+        # pylint: disable=import-outside-toplevel
+        from psyclone.psyGen import Kern
+
+        # If we find a Kern or Call child then we abort.
+        # Note that if the transformation is used it will have already
+        # attempted to do this inlining.
+        if len(self.walk((Kern, Call))) > 0:
+            raise GenerationError(
+                "Attempted to lower to OMPTaskDirective "
+                "node, but the node contains a Call or Kern "
+                "which must be inlined first."
+            )
+
+        # Create the clauses
+        (
+            private_clause,
+            firstprivate_clause,
+            shared_clause,
+            in_clause,
+            out_clause,
+        ) = self._compute_clauses()
+
+        if len(self.children) < 2 or private_clause != self.children[1]:
```

Comment:
I think because I was checking equality before making the clauses before, but I think we've moved away from that now and the clauses are created by `_compute_clauses`, so I can just replace all the original children instead and it has no difference in terms of cost.